lib/src/api.mli

```diff
@@ -55,17 +53,17 @@ module type BASIC_CHARACTER = sig
 
   val size: t -> int
   (** Get the size of the character, the exact semantics are
-      implementation-specific (c.f. {!write_to_native_string}) *)
+      implementation-specific (c.f. {!write_to_native_bytes}) *)
 
-  val write_to_native_string: t -> buf:String.t -> index:int -> (int, [> `out_of_bounds]) result
-  (** [write_to_native_string c ~buf ~index] serializes
+  val write_to_native_bytes: t -> buf:Bytes.t -> index:int -> (int, [> `out_of_bounds]) result
+  (** [write_to_native_bytes c ~buf ~index] serializes
       the character [c] at position [index] in the native string
       [buf] (writing [size c] units). Note, as with {!size} that the
       meaning of [index] is implementation dependent (can be the {i
       index-th} byte, the {i index-th} bit, etc.). *)
 
-  val to_native_string: t -> String.t
-  (** [to_native_string c] creates a string containing the
+  val to_native_bytes: t -> Bytes.t
+  (** [to_native_bytes c] creates a string containing the
```

Comment:
@smondet Checkout https://github.com/hammerlab/sosa/pull/43 and merge if you think it makes more sense, I'm starting to suspect that it does.